.github/workflows/build-lint-test.yml

```diff
@@ -53,7 +53,52 @@ jobs:
           node-version: ${{ matrix.node-version }}
           cache: yarn
       - run: yarn --immutable --immutable-cache
-      - run: yarn build
+      - name: Build TypeScript files
+        run: yarn build:tsc
+      - name: Build other files
+        run: yarn build:post-tsc:ci
+      - name: Require clean working directory
+        shell: bash
+        run: |
+          if ! git diff --exit-code; then
+            echo "Working tree dirty at end of job"
+            exit 1
+          fi
+
+  build-simulator:
+    name: Build Snaps Simulator
+    runs-on: ubuntu-latest
+    needs: prepare
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [16.x, 18.x]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: yarn
+      - run: yarn --immutable --immutable-cache
+      - name: Restore test build
```

Comment:
Can we rename this from the test build in this PR? Perhaps just call it execution-environments build everywhere?